### PR TITLE
Feature/api

### PR DESCRIPTION
**Pull request by Kanyorok**
- [x] Create an API endpoint to list all posts for a user.

- [x] Create an API endpoint to list all comments for a user's post.

- [x] Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user who makes it; it should not be set by passing a parameter.

- [x] The API endpoints receive JSON and respond to JSON as well.

### CODE REVIEW

File: app/controllers/api/v1/comments_controller.rb
Comment (shahadat3669) [inline app/controllers/api/v1/comments_controller.rb:1]: 👍 Good progress, but there's a slight adjustment needed for better functionality:

- [ ] **API Endpoint Authentication:** Currently, when requesting API endpoints for posts and comments, it prompts for signup and doesn't return the expected JSON output. I recommend updating your code to either add a way for API endpoints to authenticate requests or consider not requiring authentication specifically for API endpoints in this milestone. (only for API endpoints, remember it should not affect the previous milestone works.)

| Your | Expected|
| --- | --- |
|  image | image|


- [ ] API endpoint authentication is not mandatory for this milestone, and it's common to allow public access to certain endpoints without authentication. Consider adjusting your authentication logic to ensure that API endpoints can be accessed without the need for signup. (only for API endpoints, remember it should not affect the previous milestone works.)
- [ ] Additionally, the API endpoint for adding a new comment to a post is not working (see image). I'm getting ActionController::InvalidAuthenticityToken (Can't verify CSRF token authenticity.) when using the API endpoint to add a new comment. This error is due to rails [cross site request forgery CSRF](https://guides.rubyonrails.org/security.html#cross-site-request-forgery-csrf), there are several solution for this issue but I will suggest that you add this code skip_before_action :verify_authenticity_token to your api/comments_controller.rb

| Your | Expected|
| --- | --- |
|  image| image|


By making this adjustment, you'll provide a smoother experience for users interacting with the API. Keep up the good work! 🚀🔐

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @Kanyorok, @Al-Amin-Khan-Shakil, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ No linter Errors
✅ Everyone contributed.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by topeogunleye** [comment]
@Kanyorok, @Al-Amin-Khan-Shakil, please can you kindly make sure that your linters are running?, If you need help, please send me a DM.

**Comment by Al-Amin-Khan-Shakil** [comment]
@topeogunleye I have fixed the linters. Thank you for noticing.

**Review by topeogunleye — APPROVED** [review]
### Hi @Kanyorok, @Al-Amin-Khan-Shakil,

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Removed the authentication from api endpoint :heavy_check_mark:
 
## [Optional] suggestions:
* [ ]  Please check my inline comments.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**